src/components/Experience.tsx

```diff
@@ -31,28 +26,27 @@ const experienceDetails: ExperienceDetail[] = [
 ];
 
 const Experience = () => {
-  const [currentExperienceDataIndex, updateExperienceDataIndex] = useState<
-    undefined | number
-  >(undefined);
+  const [currentExperience, setExperience] = useState<undefined | IExperience>(
+    undefined
+  );
 
   return (
     <Grid
       templateAreas={`"timeline experienceDetails"`}
       templateRows={"1fr"}
       templateColumns={"20rem 1fr"}
       height="100%"
+      p={5}
     >
-      <GridItem area="timeline" p={5}>
+      <GridItem area="timeline">
         <ExperienceTimeline
           experienceDetails={experienceDetails}
-          updateExperienceDataIndex={updateExperienceDataIndex}
+          setExperience={setExperience}
         />
       </GridItem>
+
       <GridItem area="experienceDetails">
-        <ExperienceDetails
-          details={experienceDetails}
-          index={currentExperienceDataIndex}
-        />
+        <ExperienceDetails experienceDetails={currentExperience} />
```

Comment:
Having the prop name "experienceDetails" for both ExperienceTimeline and ExperienceDetails components where one receives the entire experience list and the other one just the current experience makes interpreting components confusing.

I think we can name the experienceDetails list as just "experiences" because it is a collection of Experience objects as can be derived from its type definition. Then, we can just name the props as "experiences" and "experience" and pass "experiences" and "currentExperience" respectively as the values.